solutions/usaco-599.mdx

```diff
@@ -31,127 +31,172 @@ a one-base polynomial hash to check if the substring appears only once. If we
 find an appropriate substring, we find the total distance that we travelled
 before finding out where we were.
 
+## Implementation
 **Time Complexity:** $\mathcal{O}(N^3)$
 
+<LanguageSection>
+<CPPSection>
+
 ```cpp
 #include <bits/stdc++.h>
-
 using namespace std;
 
-typedef pair<int, int> pii;
-#define f first
-#define s second
 using ll = long long;
 
-const ll X = 97;
-const ll M = 999999929;
+const ll P = 31;
+const ll MOD = 1e9 + 7;
+const ll MX = 1e5 * 2;
 
-int n, m, H[505], p[505];
-vector<int> k;
-pii arr[205];
+struct coord {
+	int x, y;
+};
 
-void gen_hash()
-{
-	p[0] = 1;
-	for(int i = 1; i < m; i++)
-	{
-		p[i] = int(p[i-1]*X%M);
-	}
-	H[0] = k[0];
-	for(int i = 1; i < m; i++)
-	{
-		H[i] = int((H[i-1]*X+k[i]) % M);
+signed main() {
```

Comment:
```suggestion
int main() {
```

Reply:
not committed?

Reply:
https://github.com/cpinitiative/usaco-guide/pull/3092/commits/5b69ddc95142ea333ef74de4abcd0b19e822accb

Reply:
oops i think i undo'd it